RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+// List all public photos
+const listResponse = await Storage.list({
+  key: getPrefixReference('photos/', { level: 'public' })
+})
+const firstPhoto = listResponse.files?.[0];
+
+/*
+As a note, APIs will allow customers to specify keys by string if they do not need to override the access level. For
+example, the following operation will list all files for the current user.
+*/
+const listResponseDefault = await Storage.list({
+  key: 'photos/'
+})
+
+// Copy the first photo returned to the current user's private prefix
+if (firstPhoto) {
+  await copy({
+    source: firstPhoto,
+    destination: copyObjectReference(firstPhoto, { level: 'private' }),
+  });
+}
+```
+
+## Splitting up the `get` API
+
+To better capture customer intent and simplify API types we will split up the `get` API into `getUrl` & `download`. An example for generating a pre-signed URL & downloading a file from the results of a `list` operation is highlighted below.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```TypeScript
+// List public photos
+const listResponse = await Storage.list('photos/', { level: 'public' });
+const firstPhoto = listResponse.results?.[0];
+
+// Generate a pre-signed URL for a file
+const presignedUrl = await Storage.get(firstPhoto.key, { level: 'public' });
+
+// Download a file
+const downloadResult = await Storage.get(firstPhoto.key, { download: true, level: 'public' });
+
+// Customer provided utility for handling downloaded file
+downloadBlob(downloadResult.Body, 'download.jpg');
+```
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```TypeScript
+// List public photos
+const listResponse = await Storage.list({
+  key: getPrefixReference('photos/', { level: 'public' })
+})
+const firstPhoto = listResponse.files?.[0];
+
+// Generate a pre-signed URL for a file
+const presignedUrl = await Storage.getUrl({ key: firstPhoto });
+
+// Download a file
+const downloadResult = await Storage.download({ key: firstPhoto });
+
+// Customer provided utility for handling downloaded file
+downloadBlob(downloadResult.content, 'download.jpg');
+```
+
+## Changes to the `put` return object
+
+To better capture customer intent the `put` API will be renamed to `upload`. Additionally `upload` will enable resumability by default in order to simplify API usage and remove the need to provide callbacks for monitoring upload status in favor of a Promise.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```TypeScript
+// Upload a public file with resumability enabled
+const uploadTask = Storage.put('movie.mp4', fileBlob, {
+  resumable: true,
+  level: 'public',
+  progressCallback: (progress) => {
+    // Progress of upload
+  },
+  completeCallback: (event) => {
+    // Upload finished
+  },
+  errorCallback: (err) => {
+    // Upload failed
+  }
+});
+
+// Pause & resume upload
+uploadTask.pause();
+uploadTask.resume();
+```
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```TypeScript
+// Upload a public file with resumability enabled by default
+const uploadTask = Storage.put({
```

Comment:
this still says `put` - Am I missing something? Also isn't the name of this function `uploadFile` for native?